annotation-utils/src/main/java/edu/isi/vista/annotationutils/PushAnnotations.kt

```diff
@@ -0,0 +1,177 @@
+}
+
+fun pushUpdatedAnnotations(git: Git) {
+
+    // Source: https://github.com/kittinunf/Kotlin-Playground/blob/master/buildSrc/src/main/java/com/github/kittinunf/gradle/ReleaseToMasterMergeTask.kt
+    class SshConfig(val sshKeyPath: String): TransportConfigCallback {
+        override fun configure(transport: Transport?) {
+            (transport as SshTransport).sshSessionFactory = object: JschConfigSessionFactory() {
+                override fun configure(hc: OpenSshConfig.Host?, session: Session?) {}
+                override fun createDefaultJSch(fs: FS?): JSch = super.createDefaultJSch(fs).apply {
+                    addIdentity(sshKeyPath)
+
+                }
+            }
+        }
+    }
+
+    val status = git.status().call()
+
+    // A clean status indicates no updates
+    if (!status.isClean()) {
+        val currentDate = Calendar.getInstance(TimeZone.getTimeZone("UTC"))
+        TimeZone.setDefault(TimeZone.getTimeZone("UTC"))
+        val formatter = SimpleDateFormat("yyyy-MM-dd-HH-mm-ss")
+
+        git.add().addFilepattern("data/").call()
+        git.commit()
+                .setMessage("includes annotations from ${currentDate.getTime()}").call()
+        git.tag().setName(formatter.format(currentDate.getTime())).call()
+        pushLogger.info {"Pushing updated data"}
+        git.push()
+                .setTransportConfigCallback(SshConfig("~/.ssh/id_rsa"))
+                .call()
+    } else {
+        pushLogger.info {"No updates - nothing will be pushed"}
+    }
+}
+
+fun setUpRepository(localWorkingCopyDirectory: String, repoToPushTo: String): Git {
+    /**
+     * Check if `localWorkingCopyDirectory` exists.
+     *   If it does: check the state
+     *   If not: clone `repoToPushTo` to the location specified by `localWorkingCopyDirectory`
+     */
+    if (File(localWorkingCopyDirectory).isDirectory()) {
+        val git = Git.open(File(localWorkingCopyDirectory))
+        val remoteUrl = git.getRepository()
+                .getConfig()
+                .getString("remote", "origin", "url")
+        // Does remote URL match `repoToPushTo`?
+        if (remoteUrl != repoToPushTo){
+            throw RuntimeException("$localWorkingCopyDirectory's remote URL ($remoteUrl)" +
+                    "does not match that of the remote repository ($repoToPushTo).")
+        }
+        // Is the status clean?
+        val status = git.status().call()
+        if (!status.isClean()) {
+            throw RuntimeException("$localWorkingCopyDirectory has been modified locally.")
+        }
+        // Is it on the master branch?
+        val currentBranch = git.getRepository().getBranch()
+        if (currentBranch != "master") {
+            throw RuntimeException("$localWorkingCopyDirectory is not on the master branch.")
+        }
+        // Sync to remote master
+        git.fetch().setRemote("origin").call()
+        git.reset().setRef("origin/master").setMode(ResetCommand.ResetType.HARD).call()
```

Comment:
In Git, (roughly speaking) your working copy stores its own copy of the state of the remote repository, but it only updates this copy when requested; you need to run a "fetch" here to make sure that what your working copy believes is `origin/master` truly is `origin/master`.